xdsl/irdl/attributes.py

```diff
@@ -237,12 +237,13 @@ def irdl_param_attr_definition(cls: type[_PAttrT]) -> type[_PAttrT]:
         type_index = parameter_names.index("type")
         new_fields["get_type_index"] = lambda: type_index
 
-    cls = cast(type[_PAttrT], cls)
-
     return runtime_final(
         dataclass(frozen=True, init=False)(
             type.__new__(
-                type(cls), cls.__name__, (cls,), {**cls.__dict__, **new_fields}
+                type(cls),  # pyright: ignore[reportUnknownArgumentType]
+                cls.__name__,
+                (cls,),  # pyright: ignore[reportUnknownArgumentType]
```

Comment:
these ignores are only needed for our pyright, the new one is happy with this, I'll do a pass to remove them with the automated "unnecessary ignores as errors" option